addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -172,6 +172,10 @@ var SnippetEditor = Widget.extend({
      */
     start: function () {
         var defs = [this._super.apply(this, arguments)];
+        if (!this.$target.closest('body').length) {
+            defs.push(this.destroy());
+            return Promise.all(defs);
+        }
```

Comment:
I understand the fix but I fear you fix the symptom instead of the cause: normally, in the flow you described, this should happen:

1. Drop the image gallery in the page
2. The gallery widget is created, removing the arrows
3. The gallery options are created, there is no arrow.

Or

1. Drop the image gallery in the page
2. The gallery options are created, and the arrow ones.
3. The gallery widget is created, removing the arrows
4. The arrows are not in the DOM anymore, their related option are destroyed

Whatever one it is, I don't understand where the problem actually is, I fear like one step does not wait for the full completion of the previous one.

Reply:
Following the snippet cloning process, the options of a dropped snippet are now created before starting its widgets